src/client/array/dc_array.c

```diff
@@ -2351,9 +2351,9 @@ add_record(daos_handle_t oh, daos_handle_t th, struct set_size_props *props, d_l
 	if (io_task)
 		tse_task_complete(io_task, rc);
 err:
-	D_FREE(params);
 	if (free_iod_recxs)
 		D_FREE(iod->iod_recxs);
+	D_FREE(params);
```

Comment:
good catch. thanks.
however sounds like there is still a bug and it should be like:
if (free_iod_recxs) {
		D_FREE(iod->iod_recxs);
		D_FREE(params);
}
because free_io_params_cb also registers the free for params and we don't want to free it twice if it's registered.
can also rename free_iod_recxs to just free_params to make it more clear.

Reply:
Thank you very much for your explanation and instruction!